certora/WormholeJoin.spec

```diff
@@ -0,0 +1,770 @@
+    uint256 amtToTake = amtToTake(canGenerate, gap, pendingBefore);
+    uint256 amtToGenerate = amtToGenerate(canGenerate, amtToTake, debtBefore);
+    uint256 feeAmt = feeAmt(e, canGenerate, vatLive, guid, line, debtBefore, pendingBefore, amtToTake);
+    uint256 operatorFeeAmt = operatorFeeAmt(canGenerate, operatorFee);
+
+    uint256 receiverDaiBalanceBefore = dai.balanceOf(receiverAddr);
+    uint256 operatorDaiBalanceBefore = dai.balanceOf(operatorAddr);
+    uint256 vowVatDaiBalanceBefore = vat.dai(vow());
+
+    uint256 inkBefore;
+    uint256 artBefore;
+    inkBefore, artBefore = vat.urns(ilk(), currentContract);
+
+    uint256 cureBefore = cure();
+
+    uint256 postFeeAmount;
+    uint256 totalFee;
+    postFeeAmount, totalFee = mintPending(e, guid, maxFeePercentage, operatorFee);
+
+    int256 debtAfter = debt(guid.sourceDomain);
+
+    bool    blessedAfter;
+    uint248 pendingAfter;
+    blessedAfter, pendingAfter = wormholes(hashGUID);
+
+    uint256 receiverDaiBalanceAfter = dai.balanceOf(receiverAddr);
+    uint256 operatorDaiBalanceAfter = dai.balanceOf(operatorAddr);
+    uint256 vowVatDaiBalanceAfter = vat.dai(vow());
+
+    uint256 inkAfter;
+    uint256 artAfter;
+    inkAfter, artAfter = vat.urns(ilk(), currentContract);
+
+    uint256 cureAfter = cure();
+
+    assert(to_mathint(debtAfter) == to_mathint(debtBefore) + to_mathint(amtToTake), "debt has not increased as expected");
+    assert(blessedAfter == blessedBefore, "blessed has changed when it should not happen");
+    assert(pendingAfter == pendingBefore - amtToTake, "pending has not decreased as expected");
+    assert(receiverDaiBalanceAfter == receiverDaiBalanceBefore + amtToTake - feeAmt - operatorFeeAmt, "balance of receiver did not increase as expected");
+    assert(vowVatDaiBalanceAfter == vowVatDaiBalanceBefore + feeAmt * RAY(), "balance of vow did not increase as expected");
+    assert(operatorDaiBalanceAfter == operatorDaiBalanceBefore + operatorFeeAmt, "balance of operator did not increase as expected");
+    assert(inkAfter == inkBefore + amtToGenerate, "ink has not increased as expected");
+    assert(artAfter == artBefore + amtToGenerate, "art has not increased as expected");
+    assert(canGenerate && amtToGenerate > 0 => cureAfter == artAfter * RAY(), "cure has not been updated as expected");
+    assert(!canGenerate || amtToGenerate == 0 => cureAfter == cureBefore, "cure has not stayed the same as expected");
+    assert(postFeeAmount == amtToTake - feeAmt - operatorFeeAmt, "postFeeAmount is not the expected value");
+    assert(totalFee == feeAmt + operatorFeeAmt, "totalFee is not the expected value");
+}
+
+// Verify revert rules on mintPending
+rule mintPending_revert(
+        join.WormholeGUID guid,
+        uint256 maxFeePercentage,
+        uint256 operatorFee
+    ) {
+    env e;
+
+    require(guid.amount <= max_uint128);
+    require(guid.nonce <= 0xffffffffffffffffffff);
+    require(guid.timestamp <= 0xffffffffffff);
```

Comment:
Similar comment here--the rule passes even with these commented out, so I'm not convinced they're necessary.